facet/src/main/java/com/iluwatar/facet/dragon/Dragon.java

```diff
@@ -0,0 +1,45 @@
+package com.iluwatar.facet.dragon;
+
+import com.iluwatar.facet.Attack;
+
+/**
+ * Dragon object needs to be protected, since the other objects shouldn't be
+ * allowed to edit its health. That is why it is in its own package with
+ * {@link DragonFacet} and all the functions have no access level modifiers,
+ * meaning only classes in the same package have access.
+ */
+public class Dragon {
+  private int health;
+
+  public Dragon(int health) {
+    this.health = health;
+  }
+
+  int facetedGetHealth() {
+    return health;
+  }
+
+  /**
+   * This has no access level modifier, so only other classes within
+   * the same package can access this function. This protects the knight
+   * from being able to arbitrarily change the dragon's health.
+   *
+   * @param health The new health of the dragon
+   */
+  void setHealth(int health) {
+    this.health = health;
+  }
+
+  void facetedReceiveAttack(Attack attack) {
+    switch (attack) {
+      case ARROW:
+        health -= 10;
+        break;
+      case WATER_PISTOL:
+        health -= 15;
+        break;
+      default:
+        health -= 5;
+    }
```

Comment:
I think you could improve this with an optimised switch statement. I think the syntax is similar to this:
```java
switch (attack) {
case ARROW -> {health -= 10;}
case WATER_PISTOL -> {health -= 15;}
default -> {health -= 5;}
```

Reply:
Is this legal? Maven doesn't build when I change it. I also get an error in Intellij saying that the condensed switch statements don't work in Java 11, so maybe I have to use Java 11.

Reply:
Just had a look on [this](https://docs.oracle.com/en/java/javase/13/language/switch-expressions.html) website. It seems that my syntax was pretty close. But if they don't work in Java 11 then we probably cannot make the change. I did not realise that this feature was only available on newer Java versions.